### PR TITLE
Pass network interface as command line argument

### PR DESCRIPTION
**Pull request by akhilveeraghanta**
### Description
We have way too many different interfaces, it doesn't belong in the dynamic parameters, made it a command line argument

### Testing Done
Tested manually that it works

### Resolved Issues
#1510 

### Length Justification

N/A
### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [x] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [x] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

**Review by MathewMacDougall — CHANGES_REQUESTED** [review]
Can you do this for the StandaloneSimulator main target as well? Can we as simple as an argv keyword for now, no need for boost CLI args.

**Comment by MathewMacDougall** [comment]
Also do we want to say this resolves #1511? It's definitely going to help reduce friction with interfaces, but perhaps an even more ideal solution would be an entirely virtual network interface set up in the setup scripts or something that's common for everyone? Perhaps using docker or something? May be more work than it's worth, just putting it out there.

**Comment by akhilveeraghanta** [comment]
> Also do we want to say this resolves #1511? It's definitely going to help reduce friction with interfaces, but perhaps an even more ideal solution would be an entirely virtual network interface set up in the setup scripts or something that's common for everyone? Perhaps using docker or something? May be more work than it's worth, just putting it out there.

There is no easy way to do this I have tried setting up a virtual interface with a bridge but it only works for ethernet, not on wifi

PR for that was closed here #1511 
^^ that was a script to setup a network interface and then link that with the network interfaces, worked like a charm on my vm but didn't work on Jons computer

I asked on SO and they said this was really the only way to go: 
https://stackoverflow.com/questions/62744982/how-to-make-a-program-agnostic-of-the-network-interface

**Comment by akhilveeraghanta** [comment]
> Can you do this for the StandaloneSimulator main target as well? Can we as simple as an argv keyword for now, no need for boost CLI args.

done

**Review by MathewMacDougall — DISMISSED** [review]
Cool, thanks for looking into this. Can you link your stackoverflow post into the ticket as well for reference?

**Review by kelvinkoon — DISMISSED** [review]

File: src/software/full_system_main.cpp [hunk 1]
```diff
@@ -52,14 +53,23 @@ commandLineArgs parseCommandLineArgs(int argc, char **argv)
...
         boost::program_options::value<std::string>(&args.backend_name)->required(),
-        backend_help_str.c_str())("headless",
-                                  boost::program_options::bool_switch(&args.headless),
-                                  "Run without the FullSystemGUI");
+        backend_help_str.c_str());
+    desc.add_options()(
+        "interface",
+        boost::program_options::value<std::string>(&args.network_interface_name)
+            ->required(),
+        interface_help_str.c_str());
```
Comment (jonathanlew) [inline src/software/full_system_main.cpp]: is this required even when backend is radio?

**Review by jonathanlew — CHANGES_REQUESTED** [review]

File: src/software/full_system_main.cpp (hunk 1 above)
Comment (jonathanlew) [reply src/software/full_system_main.cpp]: I think it should be an optional override, where the default is whatever is in dynamic params

**Comment by garethellis0** [comment]
crap sorry, didn't mean to press that

File: src/software/full_system_main.cpp (hunk 1 above)
Comment (garethellis0) [reply src/software/full_system_main.cpp]: I'm not sure that there's a common enough interface name for our usages to justify having a default dynamic param value for this?

**Review by garethellis0 — DISMISSED** [review]
LGTM pending existing comments! :+1: :tada:

File: src/software/full_system_main.cpp (hunk 1 above)
Comment (akhilveeraghanta) [reply src/software/full_system_main.cpp]: it has a default value already, I think Jons point was more about not specifying the interface for radio, fixed to make it optional

**Review by jonathanlew — APPROVED** [review]
LGTM! :+1:

**Review by kelvinkoon — APPROVED** [review]

**Review by MathewMacDougall — APPROVED** [review]
LGTM :+1:

**Pull request merged**